.github/workflows/update-changelog.yml

```
@@ -0,0 +1,87 @@
+    needs: tag-check
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout main
+        uses: actions/checkout@v6.0.2
+        with:
+          ref: main
+          fetch-depth: 0
+
+      - name: Find previous release tag
+        id: prev-tag
+        run: |
```

actionlint (1.7.12)

[error] shellcheck reported issue in this script: SC2086:info:11:30: Double quote to prevent globbing and word splitting

(shellcheck)